src/dataflow/mod.rs

```diff
@@ -198,9 +206,13 @@ impl Dataflow {
     ///
     /// It emits a `(key, updated_value)` tuple downstream for each
     /// input item.
-    #[pyo3(text_signature = "(self, builder, mapper)")]
-    fn stateful_map(&mut self, builder: TdPyCallable, mapper: TdPyCallable) {
-        self.steps.push(Step::StatefulMap { builder, mapper });
+    #[pyo3(text_signature = "(self, step_id, builder, mapper)")]
+    fn stateful_map(&mut self, step_id: String, builder: TdPyCallable, mapper: TdPyCallable) {
```

Comment:
the `step_id` is a name here, yes? Would it be clearer to call it `step_name` or `task_name`?

Reply:
Bytewax doesn't care what you call it, just that it's unique per "version of the dataflow." (Which is something we'll need to suss out later, how we evolve this state over time.) To me, `id` implies uniqueness more than `name`.

Reply:
Also we use in the docs `step` as a formal word to mean "a specific use of an operator with your custom logic".

Reply:
ID for a unique value makes sense.

Is uniqueness guaranteed, then? Since we're creating new tables I'm not sure if the primary key will bark.